### PR TITLE
Retrieve data from api

### PR DESCRIPTION
**Pull request by sbotichelli**
Data is being retrieved from API. Separate methods for:

- bus coordinates 
- tram coordinates

Data from bus stops:
- get a stop ID (stop name to input)
- get the list of lines for the stop (stop ID and shelter number to input. Shelter number 01, 02, etc.)
- get a timetable for the line (input stop ID, shelter number and line number/name)

ToDo: create an issue to download the latest file with bus stops list from FTP and write a parser for it. 

#3

### CODE REVIEW

**Comment by dvor** [comment]
No need to commit .DS_Store. You can add it to .gitignore and then remove from git (with `git rm --cached .DS_Store`)

File: BusTracker/BusPositionRetrieverConfig.swift [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    
+    // MARK: - Positions
+    case positionsBus
+    case positionsTram
+    
+    case stopSetId(busStopName : String)
+    case stopLineList(busStopID : String, busStopNumber : String)
+    
+    case lineTimetable(busStopId : String, busStopNumber : String, busStopLine : String)
+    //case settingsFor(cameraId: String)
```
Comment (dvor) [inline BusTracker/BusPositionRetrieverConfig.swift]: I guess this `case` and one below can be removed.

File: BusTracker/BusPositionRetrieverConfig.swift [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    
+    // 6:
+    var parameters: [String: Any] {
+        switch self {
+        case .positionsBus:
+            var parameters = [String: String]()
+            parameters["resource_id"] = API.resourceIdList["positionsResourceId"]
+            parameters["apikey"] = API.privateKey
+            parameters["type"] = "1"
+            return parameters
```
Comment (dvor) [inline BusTracker/BusPositionRetrieverConfig.swift]: You can reduce code duplication here by creating `parameters` outside of `switch`, e.g.

```objc
var parameters = [String: String]()
parameters["apikey"] = API.privateKey

switch self {
    case .positionBus:
            parameters["resource_id"] = API.resourceIdList["positionsResourceId"]
            parameters["type"] = "1"
    ....
}

return parameters
```

File: BusTracker/Networking/APIResponceParser.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    func parcePositions(vehicleType : VehicleType,APIResponce : Response) -> [TransportPosition] {
+        
+        var parcedVehiclePositions: [TransportPosition] = []
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
+        
+        if let responce = try! JSONSerialization.jsonObject(with: APIResponce.data, options: []) as? [String : Any] {
+            
+
```
Comment (dvor) [inline BusTracker/Networking/APIResponceParser.swift]: Please remove empty lines here and below. One empty line as a separator between blocks should be enough in most cases. 🙂

File: BusTracker/Networking/APIRetriever.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//  BusPositionRetriever.swift
+//  BusTracker
+//
+//  Created by botichelli on 7/12/19.
+//  Copyright © 2019 Oleksandra Sildushkina. All rights reserved.
+//
+
+import Foundation
+import Moya
+import SwiftyJSON
```
Comment (dvor) [inline BusTracker/Networking/APIRetriever.swift]: SwiftyJSON is not used

File: Podfile [hunk 5]
```diff
@@ -6,9 +6,11 @@ target 'BusTracker' do
 # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
 use_frameworks!
 
-# Pods for BusTracker
+
+
 pod 'Moya', '~> 13.0'
 pod 'SnapKit', '~> 5.0.0'
+pod 'SwiftyJSON'
```
Comment (dvor) [inline Podfile]: SwiftyJSON is not used and isn't needed for now. Please remove it from the project.

File: BusTracker/Networking/APIRetriever.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            (result) in
+            switch result {
+            case .success(let responce):
+                vehicleDataArray += self.parser.parcePositions(vehicleType: .bus, APIResponce: responce)
+            //TODO: update interface
+            case .failure(let error):
+                print(error)
+            }
+        }
+
```
Comment (dvor) [inline BusTracker/Networking/APIRetriever.swift]: Also please remove empty lines below.

File: BusTracker/Networking/APIResponceParser.swift [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        
+        // Debug
+        switch vehicleType {
+        case .bus:
+            print("Received and parced buses")
+            //print(parcedVehiclePositions)
+        case .tram:
+            print("Received and parced trams")
+            //print(parcedVehiclePositions)
+        }
```
Comment (dvor) [inline BusTracker/Networking/APIResponceParser.swift]: Is this debug code needed?

File: BusTracker.xcodeproj/xcshareddata/xcschemes/BusTracker.xcscheme [hunk 8]
```diff
@@ -71,6 +71,13 @@
             ReferencedContainer = "container:BusTracker.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
+      <EnvironmentVariables>
+         <EnvironmentVariable
+            key = "APIKey"
+            value = "1b7a5624-3dce-47c0-bcf1-0960380db9be"
```
[the remark this answers is not in the document]
Comment (sbotichelli) [reply BusTracker.xcodeproj/xcshareddata/xcschemes/BusTracker.xcscheme]: Seems to be resolved. Did I do that right?

[the remark this answers is not in the document]
Comment (dvor) [reply BusTracker.xcodeproj/xcshareddata/xcschemes/BusTracker.xcscheme]: Yes, no key in the repository now.

**Pull request merged**